pageserver/src/http/openapi_spec.yml

```
@@ -346,23 +346,23 @@ paths:
         starts writing to the tenant's S3 state unless it receives one of the
         distinguished errors below that state otherwise.
 
-        The method to identify whether a request has arrived at the pageserver, and
-        whether it has succeeded, is to poll for the tenant status to reach "Active"
-        or "Broken" state. These values are currently not explicitly documented in
-        the API spec.
-        Polling for `has_in_progress_downloads == false` is INCORRECT because that
-        value can turn `false` during shutdown while the Attach operation is still
-        unfinished.
+        If a client receives a not-distinguished response, e.g., a network timeout,
+        it MUST retry the /attach request and poll again for the tenant's
+        attachment status.
 
-        There is no way to cancel an in-flight request.
+        After the client has received a 202, it MUST poll the tenant's
+        attachment status (field `attachment_status`) to reach state `attached`.
+        If the `attachment_status` is missing, the client MUST retry the `/attach`
+        request (goto previous paragraph). This is a robustness measure in case the tenant
+        status endpoint is buggy, but the attach operation is ongoing.
```

```suggestion
        If the `attachment_status` is missing, the client MUST retry the `/attach`
        request (goto previous paragraph). This is a robustness measure in case the tenant
        status endpoint is buggy, but the attach operation is ongoing.
```
I thought we concluded that can just make `attachment_status` mandatory. And then we can just error out if it's missing, because it shouldn't happen.

---

Yeah, it is mandatory, so, pageserver would be at fault.
Yet, I feel like it's important to spell out that the client should still assume the attach has started and the pageserver is writing to the S3 prefix of the tenant.
("robustness measure", see last sentence of your quote)

You think it's not necessary / not worth the potential confusion?

---

Resolving this, @hlinnaka please open a follow-up issue if you want to improve the docs.